test/test_transformers.py

```diff
@@ -1330,6 +1330,34 @@ def test_flash_autocast_fp32_bfloat16(self):
                 _ = torch.nn.functional._scaled_dot_product_attention(
                     q, k, v, None, 0.0, False, False)
 
+    @parametrize("device", device_list)
+    @slowTest
+    def test_train_with_is_causal(self, device):
+        iters = 100
+        layer = nn.TransformerEncoderLayer(
+            d_model=2,
```

Comment:
If we want to exercise the fused kernels I would change the d_model to be 16 hence 16/2 heads = 8.  Also, the failure initially occurred for fp16 right? Maybe we should do this test with that. 

Would this test fail prior to this PR? I am not 100% what its testing, besides more of an end-to-end exercise. Not sure what failures this shows though